compiler/ast/src/types/tuple.rs

```diff
@@ -44,6 +44,14 @@ impl TupleType {
 
 impl fmt::Display for TupleType {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "({})", self.elements.iter().map(|x| x.to_string()).collect::<Vec<_>>().join(","))
+        write!(f, "(")?;
+        let mut iter = self.elements.iter().peekable();
+        while let Some(element) = iter.next() {
+            write!(f, "{element}")?;
+            if iter.peek().is_some() {
+                write!(f, ", ")?;
+            }
+        }
+        write!(f, ")")
```

Comment:
Could this be rewritten as:
```suggestion
                write!(f, "({})", self.elements.iter().map(|x| x.to_string()).collect::<Vec<_>>().join(", "))
```